### DISCUSSION TITLE
"Waiting for connection" bug

### DISCUSSION THREAD

**Discussion by 0xFloop** [rainbow-me/rainbowkit#339] [Feedback]
hey there, RainbowKit seems really awesome as i dive into it and i wanna share some things ive observed. i believe there is a bug in which connecting through metamask with RainbowKit leaves you endlessly "Waiting for connection" even though the connection has been made. Also, "disconnect"ing does not seem to actually disconnect provider from the site. If you "disconnect" then try to re connect it does not trigger metamask nor a ui update with the still connected wallet info.

Cheers and thanks for significantly improving web3 ease of use :)

![Screen Shot 2022-05-07 at 2 00 36 AM](https://user-images.githubusercontent.com/95703085/167247273-3f32a453-b49c-465c-bd8e-9b5422d4d24f.png)

**Comment by nickbytes** [comment]
@0xFloop thanks for giving RainbowKit a try. Would you mind [jumping over to the issues](https://github.com/rainbow-me/rainbowkit/issues/new?assignees=&labels=&template=bug_report.yml&title=%5Bbug%5D+%3Ctitle%3E) and following the steps there so we can help get this sorted?
